pkgs/applications/misc/nnn/default.nix

```diff
@@ -19,7 +24,11 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [ pkg-config ];
   buildInputs = [ readline ncurses ];
 
-  makeFlags = [ "DESTDIR=${placeholder "out"}" "PREFIX=" ];
+  makeFlags = [
+    "DESTDIR=${placeholder "out"}"
+    "PREFIX="
+  ] ++ optional withIcons [ "O_ICONS=1" ]
+    ++ optional withNerdIcons [ "O_NERD=1" ];
```

Comment:
`nnn` seems to have many more compile options: https://github.com/jarun/nnn/blob/master/Makefile#L14-L27. So maybe it would be better to have a `extraMakeFlags` instead.

I still end with this approach since this allowed me to do some assertions (if `O_ICONS` and `O_NERD` is both passed at the same time `O_ICONS` has preference and there is no warning or anything). And I think those two options is the most used by users (for example, in AUR there is three `nnn` packages, one for git, other for icons and other for nerd: https://aur.archlinux.org/packages/?O=0&K=nnn).